src/components/LibraryUnit.tsx

```diff
@@ -34,33 +34,20 @@ export const LibraryUnit = ({
 }) => {
   const ref = useRef<HTMLDivElement | null>(null);
   useEffect(() => {
-    const elementsToRender = elements || pendingElements;
-    if (!elementsToRender) {
-      return;
-    }
-    let svg: SVGSVGElement;
-    const current = ref.current!;
-
     (async () => {
-      svg = await exportToSvg(elementsToRender, {
+      const elementsToRender = elements || pendingElements;
+      if (!elementsToRender) {
+        return;
+      }
+      const svg = await exportToSvg(elementsToRender, {
         exportBackground: false,
         viewBackgroundColor: oc.white,
         files,
       });
-      for (const child of ref.current!.children) {
-        if (child.tagName !== "svg") {
-          continue;
-        }
-        current!.removeChild(child);
+      if (ref.current) {
+        ref.current.innerHTML = svg.outerHTML;
```

Comment:
For the life of me I couldn't figure out why appending SVG nodes containing `image` elements doesn't render the image (or set the dimensions) correctly. Repro: https://codesandbox.io/s/beautiful-black-9orux?file=/src/index.js (note: replacing `image` with `img` — which isn't valid HTML AFAIK — does work in [CSB](https://codesandbox.io/s/awesome-fast-tw2tn?file=/src/index.js), but not in Excalidraw for some reason). Using `innerHTML` setter works.

I don't think we're subject to XSS as the output of the SVG should be safe (note: we're already using `outerHTML` when saving to `.svg` file). If that's a concern, we'll need to sanitize user-supplied content at the source.

Reply:
Yeah `img` is not a valid tag, for ref [docs]( https://developer.mozilla.org/en-US/docs/Web/SVG/Element/svg#usage_notes)
I also checked this and it's weird that via JS it's not working, however, if I hardcode the data URL in the image tag it works fine